File: factorial_sum/php/factorial_sum.php
<?php

function permute($in): string {
	$total=gmp_init(1);
	for ($i=1; $i <= $in; $i++) {
		$total = gmp_mul($total, gmp_init($i));
	}
	return $total;
}

$digits = permute(100);

$total=0;
for ($i=0; $i < strlen($digits); $i++) {
	$total += substr($digits, $i, 1);
}

echo $total;

gmp_init(1);

File: factorial_sum/java/factorial_sum.java
import java.math.BigInteger;

public class factorial_sum {
	public static void main(String[] args) {
		String digits = permute(100).toString();
		int total = 0;
		for (int i=0; i<digits.length(); i++) {
			total += Integer.parseInt(digits.substring(i, i+1));
		}
		System.out.println(total);
	}

	public static BigInteger permute(int num) {
		BigInteger returnMe = BigInteger.ONE;
		for (int i=2; i<= num; i++) {
			returnMe = returnMe.multiply(BigInteger.valueOf(i));
		}
		return returnMe;
	}
}
